c_src/erlang_jq_nif.c

```diff
@@ -253,9 +269,71 @@ static ERL_NIF_TERM make_ok_return(ErlNifEnv* env, ERL_NIF_TERM result) {
     return enif_make_tuple2(env, enif_make_atom(env, "ok"), result);
 }
 
+typedef struct {
+    ErlNifMutex* lock;
+    volatile bool consumed_by_process_json_nif;
```

Comment:
In theory you might be right but but in all practical system a write to value with the bool type will be atomic. I was trying to avoid using the c11 atomics API as I thought it might not be available everywhere. But I think I want to reconsider that now since we already specify that we want c11 in the Makefile and it seems like the atomics API is available in most recent C libraries. I will change the field to [atomic_bool](https://en.cppreference.com/w/c/atomic) (this will also make it easy to fix about memory barriers below).

Reply:
I added commit [9ac00a5](https://github.com/emqx/jq/pull/36/commits/9ac00a526593cd5967f73361dc967d573d602e97) that uses atomics API which I think makes it easier to understand the code. Thanks!